python/doc/examples/calibration/bayesian_calibration/plot_gibbs_simus.py

```diff
@@ -0,0 +1,602 @@
+#    {\sim} \mathcal N_{n}(\mat{X} \theta ; \tau^{-1} \mat{I_n} +\mat{Q}^{-1}),
+#
+# We then have the following likelihood:
+#
+# .. math::
+#
+#    \mathcal L(Y|\theta, \tau) =
+#    (2\pi)^{-\frac{n}{2}}
+#    {\rm det}(\tau^{-1}I_n+\mat{Q}^{-1})^{-\frac{1}{2}}
+#    e^{-\frac{1}{2}
+#    ||Y - \mat{X} \theta ||_{\tau^{-1}I_n+Q^{-1}}^2
+#    }
+#
+# where :math:`||X-Y||_{V}^2` is the Mahalanobis distance between
+# :math:`X` and :math:`Y` with covariance matrice :math:`V`:
+#
+# .. math::
+#
+#    ||X-Y||_{V}^2 = (X-Y)^\top\ V^{-1}\ (X-Y).
+#
+# 1.2. Interval Censorship
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We now assume that, instead of observing directly the
+# :math:`Y_i, i=1,\ldots,n` as described above, we only have access to
+# discretized values :math:`Y_i^{obs} \in \delta \times \mathbb N`, where
+# :math:`\delta > 0` is a grid length and :math:`Y_i^{obs} = \delta \times k` means that
+# :math:`Y_i \in \delta \times \left[ k - \frac{1}{2}; k + \frac{1}{2} \right[`.
+#
+# 1.3. Remarks
+# ~~~~~~~~~~~~
+#
+# -  The presence of a composite matrix :math:`\tau^{-1}I_n+i\mat{Q}^{-1}` makes
+#    estimation more complex than with a spherical one
+#    :math:`\tau^{-1}\mat{I_n},` since we would then have explicit (closed-form)
+#    maximum likelihood estimators, and also conjugate priors leading to
+#    explicit full joint posterior distributions.
+# -  Another difficulty is the presence of censored data, since the
+#    likelihood is no more available in closed-form. As we will see, this
+#    is easily overcome thanks to Bayesian inference.
+# -  Heteroscedastic linear modeling under interval censorship as
+#    formulated above was originally motivated by an industrial case-study
+#    in seismology, wherein the :math:`Y_i` correspond to the observed
+#    intensity of an earthquake in a distant site, and explanatory
+#    variables :math:`X_i` are derived from the epicentral distance to the
+#    earthquake's source as well as its characteristics (magnitude,
+#    depth). But it can also arise in many different contexts, as soon as
+#    observations are available with known statistical precisions (hence
+#    the heteroscedasticity) and limited numerical accuracy (hence the
+#    censorship).
+
+# %%
+# 1.4 Simulate the dataset
+# ~~~~~~~~~~~~~~~~~~~~~~~~
+import openturns as ot
+from openturns.viewer import View
+ot.Log.Show(ot.Log.NONE)
+ot.RandomGenerator.SetSeed(11123)
+from matplotlib import pyplot as plt
+import numpy as np
+np.random.seed(456)
+
+n = 10
+delta = 0.5
+
+
+X = np.array(ot.ComposedDistribution([ot.Dirac(1.0), 10.0 * ot.Normal()]).getSample(n))
+
+# %%
+# Make the precision matrix :math:`\mat{Q}` a diagonal matrix and sample
+# its diagonal coefficients from an :class:`~openturns.Exponential` distribution.
+Q = np.ones((n,1)) + ot.Exponential().getSample(n)
+
+
+# %%
+# Choose values for the parameters :math:`\theta` and :math:`\tau`.
+theta_true = np.array([0.0, 1.0]).reshape(-1,1)
+p = len(theta_true)
+tau_true = .1
```

Comment:
Please use Black on the script. I think it should turn ".1" into "0.1" in this instance.